docs/manually-create-secrets.md

```diff
@@ -0,0 +1,40 @@
+# Kubernetes Secrets
+
+As of Kubernetes v1.24, secrets are no longer automatically created for service accounts.
+
+You must create the
+manually. [Find out how to create these yourself manually](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#manually-create-a-service-account-api-token)
+.
+
+You must make the secret discoverable. You have two options:
+
+## Option 1 - Discovery By Name
+
+Name your secret `${serviceAccountName}.service-account-token`.
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: default.service-account-token
+  annotations:
+    kubernetes.io/service-account.name: default
+type: kubernetes.io/service-account-token
+```
+
+This option is simpler than option 2, as you can combine creating the secret with making it discoverable by name.
+
+## Option 2 - Discovery By Annotation
+
+Annotate the service account with the secret name:
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: default
+  annotations:
+    kubernetes.io/service-account-token.name: my-token
```

Comment:
Please don't use non-standard kubernetes.io-namespaced annotations. If this is the annotation argo will look for to find the secret, please use an argo-specific annotation name to hold this

Reply:
This is a standard one:

https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#manually-create-a-service-account-api-token

Reply:
that is `kubernetes.io/service-account.name`, not `kubernetes.io/service-account-token.name`, and is [defined for use on Secret objects](https://kubernetes.io/docs/reference/labels-annotations-taints/#kubernetes-io-service-account-name) pointing back at the service account, not the other way around

Reply:
Oh. You’re right. It should be `workflows.argoproj.io/service-account-token.name`. Which makes me wonder how any tests are passing.

Reply:
Fixed. It’s only wrong in the docs.